### PR TITLE
Sp4 361 not open google map navigation route to the selected contact on i os fe

### PR DESCRIPTION
**Pull request by serhiibets**

### CODE REVIEW

File: wayat/frontend/lib/services/google_maps_service/google_maps_service.dart [hunk 1]
```diff
@@ -5,16 +5,18 @@ import 'dart:io';
...
 
 class GoogleMapsService {
   static void openMaps(double lat, double lng) async {
-    var uri = Uri.parse("google.navigation:q=$lat,$lng&mode=d");
-    if (await canLaunchUrl(uri)) {
-      await launchUrl(uri);
-    } else {
-      throw 'Could not launch ${uri.toString()}';
+    var uri;
+    if (Platform.isAndroid) {
```
[the remark this answers is not in the document]
Comment (serhiibets) [reply wayat/frontend/lib/services/google_maps_service/google_maps_service.dart:13]: Fix

File: .gitignore [hunk 2]
```diff
@@ -131,3 +131,4 @@ app.*.symbols
 
 wayat/.env
 SPIKES/firestore/firestore_example/android/app/google-services.json
+wayat/frontend/ios/Runner.xcodeproj/xcshareddata/xcschemes/Runner.xcscheme
```
[the remark this answers is not in the document]
Comment (serhiibets) [reply .gitignore:134]: Ok

File: wayat/frontend/pubspec.yaml [hunk 3]
```diff
@@ -65,6 +65,7 @@ dependencies:
   image_picker: ^0.8.5+3
   # Testing mocks
   mockito: ^5.3.0
+  smart_auth: ^1.0.5
```
[the remark this answers is not in the document]
Comment (serhiibets) [reply wayat/frontend/pubspec.yaml:68]: I have error in this file generated_plugin_registrant.dart, when I resolve it - Flutter propose me to add this dependency.

File: wayat/frontend/lib/services/google_maps_service/google_maps_service.dart (hunk 1 above)
Comment (capf4bbo) [inline wayat/frontend/lib/services/google_maps_service/google_maps_service.dart:13]: Web condition is missing

File: wayat/frontend/lib/main.dart [hunk 4]
```diff
@@ -13,13 +13,16 @@ import 'package:wayat/features/onboarding/controller/onboarding_controller.dart'
...
 import 'package:timeago/timeago.dart' as timeago;
 
 Future main() async {
-  await dotenv.load(fileName: ".env");
   WidgetsFlutterBinding.ensureInitialized();
-  await Firebase.initializeApp();
+  // Env file should be loaded before Firebase initialization
+  await dotenv.load(fileName: ".env");
+  await Firebase.initializeApp(
+      options: DefaultFirebaseOptions.currentPlatformOptions);
```
Comment (capf4bbo) [inline wayat/frontend/lib/main.dart:25]: Please name DefaultFirebaseOptions as CustomFirebaseOptions and add name argument to WAYAT in te initializeApp method

**Review by capf4bbo — CHANGES_REQUESTED** [review]

File: wayat/frontend/lib/services/google_maps_service/google_maps_service.dart (hunk 1 above)
Comment (serhiibets) [reply wayat/frontend/lib/services/google_maps_service/google_maps_service.dart:13]: Fixed

File: wayat/frontend/lib/main.dart (hunk 4 above)
Comment (serhiibets) [reply wayat/frontend/lib/main.dart]: Fixed

File: wayat/frontend/lib/main.dart [hunk 5]
```diff
@@ -13,13 +13,16 @@ import 'package:wayat/features/onboarding/controller/onboarding_controller.dart'
...
 import 'package:wayat/services/common/http_provider/http_provider.dart';
 import 'package:timeago/timeago.dart' as timeago;
 
 Future main() async {
-  await dotenv.load(fileName: ".env");
   WidgetsFlutterBinding.ensureInitialized();
-  await Firebase.initializeApp();
+  // Env file should be loaded before Firebase initialization
+  await dotenv.load(fileName: ".env");
+  await Firebase.initializeApp(
```
Comment (capf4bbo) [inline wayat/frontend/lib/main.dart:24]: Change it to:
await Firebase.initializeApp(
    name: "WAYAT",
    ...

Comment (capf4bbo) [inline wayat/frontend/lib/main.dart:24]: Change it to:
await Firebase.initializeApp(
name: "WAYAT",
...

File: wayat/frontend/lib/main.dart [hunk 6]
```diff
@@ -13,13 +13,16 @@ import 'package:wayat/features/onboarding/controller/onboarding_controller.dart'
...
+      options: CustomFirebaseOptions.currentPlatformOptions);
```
Comment (capf4bbo) [inline wayat/frontend/lib/main.dart:25]: Change it to:
await Firebase.initializeApp(
name: "WAYAT",
...

**Review by capf4bbo — CHANGES_REQUESTED** [review]

File: wayat/frontend/lib/main.dart (hunk 6 above)
Comment (serhiibets) [reply wayat/frontend/lib/main.dart]: Fixed

**Review by capf4bbo — APPROVED** [review]

**Pull request merged**